scala-package/infer/src/main/scala/ml/dmlc/mxnet/infer/ImageClassifier.scala

```diff
@@ -44,12 +45,19 @@ class ImageClassifier(modelPathPrefix: String,
 
   val classifier: Classifier = getClassifier(modelPathPrefix, inputDescriptors)
 
+  require(inputDescriptors.head.shape.length != 0,
+    "Please provide shape information in the descriptor")
+
+  require(!inputDescriptors.head.layout.isEmpty,
+    "Please provide layout information in the descriptor")
+
   val inputLayout = inputDescriptors(0).layout
 
   val inputShape = inputDescriptors(0).shape
 
   // Considering 'NCHW' as default layout when not provided
   // Else get axis according to the layout
+  // [TODO] if layout is different
```

Comment:
if layout is different than the bufferedImage layout, transpose to match the inputdescriptor shape